lib/client.js

```diff
@@ -116,7 +119,10 @@ function Client (broker, conn, req) {
     }
     const toForward = dedupe(that, _packet) &&
       that.broker.authorizeForward(that, _packet)
-    if (toForward) {
+
+    const isNL = _packet.clientId === that.id && _packet.nl
+
+    if (toForward && !isNL) {
```

Comment:
Could you create a `shouldForward` method for this as it's the same for deliver0 and deliverQos?

Reply:
Should be done now 👍